src/python/pants/backend/python/goals/setup_py.py

```diff
@@ -862,6 +862,15 @@ async def get_requirements(
     direct_deps_tgts = await MultiGet(
         Get(Targets, DependenciesRequest(tgt.get(Dependencies))) for tgt in owned_by_us
     )
+    direct_deps_chained = OrderedSet(itertools.chain.from_iterable(direct_deps_tgts))
+    # Issue #17593: If ET has an undeclared requirement on R which has been resolved with an
+    # explicit override dependency, we must preserve the explicit dependency on R in favour of the
+    # broken ET.
+    transient_3rd_party_tgts = await MultiGet(
```

Comment:
I suggest "explicit_direct_deps". These aren't transitive! You're only going one level deep.

Reply:
Although, now that I think of it, do we want to do this transitively? What if R was also missing a req that we added manually?

Reply:
yes, I believe we do want to do it transitively, to ensure we have a complete set of requirements in the end.